proposals/devcontainer-distribution.md

```diff
@@ -0,0 +1,71 @@
+- Users can pin a particular version of a feature to allow consistent repeatable environments.
+
+# Proposal
+
+## Github Topics
+
+This can be achieved with the use of [GitHub Topics](https://docs.github.com/en/repositories/managing-your-repositorys-settings-and-features/customizing-your-repository/classifying-your-repository-with-topics), specifically the [devcontainers](https://github.com/topics/devcontainers) topic. Additionally the repositories can be marked with **devcontainers-templates** and/or **devcontainers-features** to mark specifically the contents of the repository.
+
+Any UI application can leverage this search to provide a list of repositories that contain features or templates and present them for their users. The application would query the latest release and obtain the details to show them to the users.
+
+An implementation of the spec can also use other mechanisms to speed up or improve the user experience while searching.
+
+At the same time it should contain a way for users to directly reference a repository containing the features/templates that they want to use.
+
+## Releases
+
+As a distribution and storage mechanism we propose using GitHub releases. The releases would contain compressed files with the source code of the feature or template. Together with the release the corresponding metadata file would be included either **devcontainer-feature.json**, **devcontainer-template.json** or **devcontainer-collection.json**.
+
+## Collections
+
+Features and/or templates can be distributed both as a single unit or as a **collection** of them. In case they are released as a collection the release should include a **devcontainer-collection.json** file that groups the information of each feature/template contained.
+
+| Property | Type | Description |
+| :--- | :--- | :--- |
+| publisher | string | The organization/user that publishes this group of templates/features. Must match with the repository user in github.|
+| repository | string | The repository where this collection is contained.|
+| version | string | Version information of this collection list.|
+| features | array | The list of features that are contained in this collection. Same information as the metadata file for the feature.|
+| templates | array | The list of templates contained in this collection. Same information as the metadata file for the definition.|
+
+Its important to note the following about collections:
+
+- It is recomended that they are created automatically as part of the release process.
+- Ids would get the collection repository added to them. (e.g. for 'https://github.com/community/features' then 'community/features' to 'myfeature1' for a feature with that id)
+- The links to the release files would be added.
+- Date.
+- Version.
+
+## Templates specifics
+
+By their nature templates are single use since they are just a group of files that are copied over to an end users repository and used to kickstart a **dev container** configuration.
```

Comment:
```suggestion
By their nature, templates are single use since they are just a group of files that are copied over to an end user's repository and used to kickstart a **dev container** configuration.
```

Reply:
Does "single use" mean the template is only pulled once?